packages/app-mobile/android/app/src/main/cpp/utils/WhisperSession.cpp

```diff
@@ -208,6 +211,21 @@ WhisperSession::transcribeNextChunkNoPreview_() {
 	return result.str();
 }
 
+void WhisperSession::updatePreview_() {
+	if (previewEnabled_) {
+		auto start = std::chrono::steady_clock::now();
+		previewText_ = transcribe_(audioBuffer_, audioBuffer_.size());
+		auto stop = std::chrono::steady_clock::now();
+
+		// If it takes a long time to generate the preview, disable it.
+		if (stop - start >= 2s) {
```

Comment:
At present, the chosen threshold is two seconds. If it takes longer than this to generate the preview (which also delays generating the next preview), the preview box is disabled for the remainder of the voice typing session.